### DISCUSSION TITLE
shared primitives for component libraries

### DISCUSSION THREAD

**Discussion by stefanprobst** [solidjs-community/solid-primitives#272] [General]
hi, what do you think about including primitives in this repo which could be shared by solidjs component libraries, i'm thinking of things like focus trap, scroll lock, marking subtrees as [inert](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inert) etc.?

**Comment by thetarnav** [comment]
One time we wondered about this and the solid-aria project was born. As it turned out to be more about porting react-aria API than making individual primitives with specific goals, like the ones in this library.
So in the end we are still adding ui-related primitives here when we need them.
If you happen to write primitives that solve a specific issue like the ones you mentioned, we would happily accept a pr. I'm sure more people visit this repo looking for focus trap / scroll lock utilities.
